### PR TITLE
First draft for Github action

### PR DESCRIPTION
**Pull request by benediktahrens**
This is a very naive first draft for a github action. It is based on the installation instructions for Debian/Ubuntu, and I have not tested them - I would not know how to test them.

relates to #1324

### CODE REVIEW

File: .github/workflows/build-unimath.yml [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+
+      # Runs a set of commands using the runners shell
+      - name: Run a multi-line script
+        run: |
+          echo Add other actions to build,
+          echo test, and deploy your project.
+      
+      # Install everything needed for building UniMath (and more)
+      - name: Install build dependencies
+        run: sudo apt-get install build-essential git ocaml ocaml-nox ocaml-native-compilers camlp5 libgtk2.0 libgtksourceview2.0 liblablgtk-extras-ocaml-dev ocaml-findlib emacs
```
Comment (benediktahrens) [inline .github/workflows/build-unimath.yml:40]:
```suggestion
        run: sudo apt-get install build-essential git ocaml ocaml-nox ocaml-native-compilers camlp5 libgtk2.0 libgtksourceview2.0 liblablgtk-extras-ocaml-dev ocaml-findlib libnum-ocaml-dev emacs
```

File: .github/workflows/build-unimath.yml [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+
+      # Runs a single command using the runners shell
+      - name: Run a one-line script
+        run: echo Hello, world!
+
+      # Runs a set of commands using the runners shell
+      - name: Run a multi-line script
+        run: |
+          echo Add other actions to build,
+          echo test, and deploy your project.
```
Comment (benediktahrens) [inline .github/workflows/build-unimath.yml:36]:
```suggestion
```

**Comment by benediktahrens** [comment]
@DanGrayson : 

Maybe an improvement would be to split the current job "build" into two, one for doing the sanity checks, and one for the actual building? 

Running the sanity checks builds Coq, for some reason, and I am not sure if the Coq built for the sanity checks would be persistent across jobs and be available for the "build" job. Worth checking in case we split the jobs.

**Comment by benediktahrens** [comment]
Here is more explanation of jobs depending on other jobs: https://docs.github.com/en/actions/learn-github-actions/managing-complex-workflows

We do not want the "build" job to depend on the "sanity check" job: sanity-checking includes checking for changes in Foundations, which we do want to allow in some cases.

Different jobs that are not dependent on each other run in parallel, so using the Coq built during job "sanity check" is not possible when running job "build".

We could make the sanity check depend on "build", but it seems a good idea to run both checks on each PR/commit in any case. So maybe the current setup is the best?

**Comment by DanGrayson** [comment]
Having 1 job should be enough.   Why don't we just put the sanity check last?  Then the proofs will get checked, too, even if the sanity test doesn't succeed?  The proof checking step will build coq.

**Comment by benediktahrens** [comment]
> Having 1 job should be enough. Why don't we just put the sanity check last? Then the proofs will get checked, too, even if the sanity test doesn't succeed? The proof checking step will build coq.

Yes, good idea - done in the latest commit.

**Comment by benediktahrens** [comment]
@DanGrayson : is this ready for being merged? Once this is done, I'll deactivate travis on this repository.

**Comment by DanGrayson** [comment]
Something went wrong here:

https://github.com/UniMath/UniMath/pull/1352/checks?check_run_id=2738595713#step:5:12

**Comment by DanGrayson** [comment]
I'll fix that later...

**Comment by DanGrayson** [comment]
I think I've disabled travis now, too.

**Pull request merged**